src/main/java/Duke.java

```diff
@@ -1,10 +1,187 @@
+                        continue;
+                    }
+                    if (tasks.get(index-1).isDone()) {
+                        System.out.println("Already done!");
+                        input = sc.nextLine();
+                        continue;
+                    }
+                    System.out.println("Nice! I've marked this task as done:");
+                    tasks.get(index-1).markAsDone();
+                    System.out.print(tasks.get(index-1).getStatusIcon() + " ");
+                    System.out.println(tasks.get(index-1).getDesc() + " ");
+                } else if (input.startsWith("todo")) {
+                    System.out.println("Got it. I've added this task:");
+                    addTodo(input.substring(5));
+                    System.out.println("Now you have " + tasksCount + " tasks in the list");
+                } else if (input.startsWith("deadline")) {
+                    System.out.println("Got it. I've added this task:");
+                    int slashIndex = input.indexOf("/");
+                    addDeadline(input.substring(9, slashIndex - 1), input.substring(slashIndex + 1));
+                    System.out.println("Now you have " + tasksCount + " tasks in the list");
+                } else if (input.startsWith("event")) {
+                    System.out.println("Got it. I've added this task:");
+                    int slashIndex = input.indexOf("/");
+                    addEvent(input.substring(6, slashIndex - 1), input.substring(slashIndex + 1));
+                    System.out.println("Now you have " + tasksCount + " tasks in the list");
+                } else if (input.equals("list")) {
+                    System.out.println("Here are the tasks in your list: ");
+//                printTasksList();
+                    printTodoEventDeadlineList();
+                } else if(input.startsWith("delete")){
+                    String index_string = input.substring(7);
+                    int i = Integer.parseInt(index_string);
+                    System.out.println("Noted. I've removed this task: ");
+                    System.out.print(tasks.get(i-1).typeChar()+ " ");
+                    System.out.print(tasks.get(i-1).getStatusIcon() + " ");
+                    if(tasks.get(i-1) instanceof Deadline || tasks.get(i-1) instanceof Events){
+                        System.out.print(tasks.get(i-1).getDesc() + " ");
+                        System.out.println("("+ tasks.get(i-1).getBy() + ") ");
+                    }else{
+                        System.out.println(tasks.get(i-1).getDesc() + " ");
+                    }
+                    tasks.remove(i-1);
+                    tasksCount-=1;
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                }else {
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                }
+                input = sc.nextLine();
```

Comment:
The main method looks cluttered - consider abstracting some of the code into separate classes/methods.